lib/tasks/after_tasks.rake

```diff
@@ -631,6 +631,25 @@ namespace :After do
     end
     puts && STDOUT.flush
   end
+
+  desc "Update each user's kudos with the user's id"
+  task(add_user_id_to_kudos: :environment) do
+    total_users = User.all.size
+    total_batches = (total_users + 999) / 1000
+    puts "Updating #{total_users} users' kudos in #{total_batches} batches"
+
+    User.find_in_batches.with_index do |batch, index|
+      batch_number = index + 1
+      progress_msg = "Batch #{batch_number} of #{total_batches} complete"
+      batch.each do |user|
+        Kudo.where(pseud_id: user.pseud_ids)
```

Comment:
You can specify both conditions in one where: `.where(pseud_id: user.pseud_ids, user_id: nil)`. Same SQL queries but shorter code.